### DISCUSSION TITLE
Error Typed property Illuminate\Events\CallQueuedListener::$job must not be accessed before initialization

### DISCUSSION THREAD

**Discussion by trevorpan** [laravel/framework#34571] [General]
- Laravel Version: 8.6.0 (upgraded from 7.25)
- PHP Version: 7.4.10
- Database Driver & Version: MySQL 8.0.21
- Horizon 5.0

### Description:
Have finished going through Mohamed's awesome queue book. And started running into some strange errors.

I started with this:
```
class PublicMessagesMail extends Mailable implements ShouldQueue
{
    use InteractsWithQueue, Queueable, SerializesModels;

    // when these below are uncommented they give errors

    // public $tries = 3; 
    // public $timeout = 40;
    // public $retryAfter = [60, 120];
    // public $delay = 10;
    // public $connection = 'redis';
    // public $queue = 'notifications';

    protected $biddedJob;


    /**
     * The user instance.
     *
     * @var  \Illuminate\Contracts\Auth\Authenticatable
     */
    public $user;

    /**
     * Create a new message instance.
     * Set the mail queue configuration
     *
     * @param \Illuminate\Contracts\Auth\Authenticatable  $user
     */
    public function __construct($user)
    {
        $this->user = $user;

        $this->connection = 'redis';
        $this->tries = 3;
        $this->retryUntil = 40;
        $this->backoff  = [60, 120];
        $this->delay = 10;
        $this->queue = 'mail';
    }
```
I read through the https://stitcher.io/blog/typed-properties-in-php-74 post and much of RFC. Nothing I was doing seemed strange.

Then I received: `Symfony\Component\ErrorHandler\Error\FatalError App\Listeners\SendPublicMessageNotification and Illuminate\Bus\Queueable define the same property ($connection) in the composition of App\Listeners\SendPublicMessageNotification. However, the definition differs and is considered incompatible. Class was composed` 

So I found this post:https://github.com/laravel/nova-issues/issues/557#issuecomment-422963978 Taylor mentions using the constructor to set queue settings. But I get these type of errors: 

`Error Typed property Illuminate\Events\CallQueuedListener::$job must not be accessed before initialization`

### Steps To Reproduce:
I don't think I can give perfect steps to reproduce. (maybe create a App\Models\SlugJob = $job in a fresh app and then upgrade from 7-8, then see if `trait InteractsWithQueue` has the path in the vendor directory?)

After trying a few forum posts with nobody answering I dug into the trait: `Illuminate\Events\CallQueuedListener`
I found that the vendor directory had pulled in ` \App\Models\SlugJob $job;` This is very bizarre because when I upgraded I did it manually for the models directory. e.g. open each file and place `\App\Models\model_name` vs. `\App\model_name' for the namespace.

So, I **did not use** phpstorm to find and replace. 

Yet, here the framework pulled in my application `SlugJob` model. I'm not sure if this is because it uses the variable `$job`?

```
trait InteractsWithQueue
{
    /**
     * The underlying queue job instance.
     *
     * @var \Illuminate\Contracts\Queue\Job
     */
    public \App\Models\SlugJob $job;
```
After seeing that, I removed the namespace and the queues worked as expected. This also allowed me to declare the values of `$tries = 3;` for example.  As shown in the book.

So, the the app model `$job` was being accessed in the models directory and inserted, wrongly - now the error makes sense. 

```
trait InteractsWithQueue
{
    /**
     * The underlying queue job instance.
     *
     * @var \Illuminate\Contracts\Queue\Job
     */
    public $job;
```

Could this be a bug?

**Comment by taylorotwell** [comment]
I think your $job property is conflicting.
